### PR TITLE
Annotate `@Generated` to generated metamodel classes

### LINKED ISSUE #1916: Mark metamodel classes @Generated

### PR DESCRIPTION
**Pull request by MysterAitch**
Closes #1916

### CODE REVIEW

**Comment by MysterAitch** [comment]
Oh, just realised that this is based on #2554 -- this link will give the relevant diff for this issue:

https://github.com/javaparser/javaparser/pull/2556/files/365d521b1e642b67df1600e56c53e8f578da6243..b61a1f2db67b98900489e714ff7b70c881efe19d

File: javaparser-core-generators/src/main/java/com/github/javaparser/generator/core/other/TokenKindGenerator.java [hunk 1]
```diff
@@ -60,7 +60,7 @@ public TokenKindGenerator(SourceRoot sourceRoot, SourceRoot generatedJavaCcSourc
         final EnumDeclaration kindEnum = javaToken.findFirst(EnumDeclaration.class, e -> e.getNameAsString().equals("Kind")).orElseThrow(() -> new AssertionError("Can't find class in java file."));
 
         kindEnum.getEntries().clear();
-        this.annotateGenerated(kindEnum);
+        AbstractGenerator.annotateGenerated(kindEnum, this.getClass().getName());
```
Comment (matozoid) [inline javaparser-core-generators/src/main/java/com/github/javaparser/generator/core/other/TokenKindGenerator.java:63]: Yeah, those methods should be moved out of AbstractGenerator, or this one should become an AbstractGenerator.

File: javaparser-core-metamodel-generator/src/main/java/com/github/javaparser/generator/metamodel/InitializePropertyMetaModelsStatementsGenerator.java [hunk 2]
```diff
@@ -38,22 +40,27 @@
...
+
         final String propertyInitializer = f("new PropertyMetaModel(%s, \"%s\", %s.class, %s, %s, %s, %s, %s)",
                 nodeMetaModelFieldName,
                 field.getName(),
                 typeName,
                 optionalOf(decapitalize(nodeMetaModelName(fieldType)), isNode(fieldType)),
-                isOptional(field),
-                isNonEmpty(field),
+                this.isOptional(field),
+                this.isNonEmpty(field),
```
Comment (matozoid) [inline javaparser-core-metamodel-generator/src/main/java/com/github/javaparser/generator/metamodel/InitializePropertyMetaModelsStatementsGenerator.java:60]: The next time I change something here, the explicit `this.`'s will suddenly disappear again :-)

File: javaparser-core-metamodel-generator/src/main/java/com/github/javaparser/generator/metamodel/MetaModelGenerator.java [hunk 3]
```diff
@@ -21,244 +21,207 @@
...
-        " *\n" +
-        " * JavaParser is distributed in the hope that it will be useful,\n" +
-        " * but WITHOUT ANY WARRANTY; without even the implied warranty of\n" +
-        " * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the\n" +
-        " * GNU Lesser General Public License for more details.\n" +
-        " ";
+    static final String BASE_NODE_META_MODEL = "BaseNodeMetaModel";
+    static final String METAMODEL_PACKAGE = "com.github.javaparser.metamodel";
 
+    // Manually maintained..?
```
Comment (matozoid) [inline javaparser-core-metamodel-generator/src/main/java/com/github/javaparser/generator/metamodel/MetaModelGenerator.java:45]: Yes, the order here is important. Please don't automate it, that's really not worth the effort.

File: javaparser-core/src/main/java/com/github/javaparser/metamodel/AnnotationExprMetaModel.java [hunk 4]
```diff
@@ -21,17 +21,32 @@
...
 import com.github.javaparser.ast.Node;
 
+/**
+ * This file, class, and its contents are completely generated based on:
+ * <ul>
+ *     <li>The contents and annotations within the package `com.github.javaparser.ast`, and</li>
+ *     <li>`ALL_NODE_CLASSES` within the class `com.github.javaparser.generator.metamodel.MetaModelGenerator`.</li>
+ * </ul>
+ *
+ * For this reason, any changes made directly to this file risk being overwritten.
```
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/metamodel/AnnotationExprMetaModel.java:35]: Not just risk, they get definitely overwritten :-)

**Comment by MysterAitch** [comment]
Replaced by #2706 which excludes all the extra bits this PR has.

**Pull request closed without merging**